accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,202 @@
+
+At the time of writting this RFC, most of the modules dropped from the Salt core are not yet migrated to Salt Extensions, therefore nobody from the community has claimed their ownership yet.
+
+### Builtin Salt Extensions
+
+There are a couple of Salt Extensions that we want to have integrated by default in our main `python311-salt` package. These are:
+- zypperpkg
+- dpkg
+- transactional_update
+- openscap
+- docker / dockerng
+- virt (needed by virtualization-formula)
+- ...
+
+These are the minimum required extensions to be able to do basic operations in SUSE/openSUSE distributions and also allow basic operations on the context of SUSE Manager clients. We must maintain and support these builtin Salt Extensions.
+
+This list could be reduced even more by just providing the `zypperpkg` and `transactional_update` modules and then providing the rest via `/usr/share/susemanager/salt/[_modules,_states,...]` in the Uyuni / SUSE Manager server.
+
+Builtin Salt Extensions will be integrated into our main "openSUSE/salt" GitHub repo codebase for 3008.x, and provided as content of the main `python311-salt` package.
+
+#### An example of the Git repo:
+```
+salt
+|-- builin-extensions
+|   |-- saltext-zypperpkg
+|   |-- saltext-dpkg
+|   |-- ...
+...
+```
+
+The idea would be to include the releases for the different builtin extension directly inside our GH repo, as it was previously done with "tornado". We should take this oportunity also to take the ownership, migrate and create the official `saltext-zypperpkg` extension and probably some others.
+
+These builtin extensions will be then placed in `/usr/lib/python3.11/site-packages/saltext/` when installing `python311-salt` so they are available for Salt.
+
+**NOTE:** Depending on the maintaining workflow to use, the usage of `git submodule` would make it a bit tricky when it comes to generate patches (i.a. during bugfixing) for the single unified Salt codebase, as apparentely you cannot get an unified diff for the main repo + submodules.
+
+Alternatively, we could create separated packages for each of those default Salt Extensions and then make add them as `Requires` for the `python311-salt` package. In that case, an official Salt Extensions must be created (if not existing) for each extension we want to have a package, and then taking the sources from the official extension repository. One Salt Extension, one RPM package.
+
+### What about Salt Extensions packages?
+
+Essentially there are some different approaches here:
+
+1. Creating separated packages for each one of the Salt Extensions (potentially hundreds) -> maybe too cumbersome if we go with all of them, and probably not really needed.
+2. Do not package Salt Extensions at all -> builtin extensions + manual customization.
+3. Only package a reduced list of Salt Extensions -> we could consider some popular extensions that we should then maintain in the context of SLE.
```

Comment:
Having an RPM package for every potential Salt Extensions that a customer may use in their states is probably not realistic, so we definetely need a way for users and customers to be able to extend their available modules.

In this regard, in case a Salt Extension is not packaged as RPM, the users could use any of these possibilities:

- Installing extensions directly from PyPI using `salt-pip` or `pip.installed` state to install `saltext-*` packages. This would enable users to extend their minion by applying an state, so easy to integrate with Uyuni but requiring access to PyPI from the minion side.

- Providing Salt Extensions as custom modules for Salt via `/srv/salt/` from the Uyuni server.

Reply:
Thank you for the insides. Option 1 is not feasible, since we cannot maintain a 1-to-1 compatibility to all extension.
I think we can move forward as is now in your proposal, but I'm expecting we will have some requirements for 5.2 to improve content consumption.